python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,1138 @@
+
+        Returns
+        -------
+        mod : tvm.relay.Module
+            The module that optimizations will be performed on.
+
+        params : dict of str to tvm.ndarray
+            Dict of converted parameters stored in tvm.ndarray format
+        """
+        # Check for missing ops
+        missing_operators = self._parse_import_prerequisites()
+
+        if missing_operators:
+            raise NotImplementedError( \
+                "The following operators are not implemented: {}".format(missing_operators))
+
+        # Translate PyTorch graph to by decorating Graph with state dict and inputs into each op
+        self._parse_inputs()
+        self._parse_params()
+        self._parse_ops()
+
+        nid = 0
+        for (op_name, operator), op_node in self._ops.items():
+            if operator == 'prim::Constant':
+                pass
+            elif operator == 'prim::ListConstruct':
+                if any(inp.debugName() in self._nid_to_node_name.keys() \
+                       for inp in op_node.inputs()):
+                    listconstr = []
+                    for i in op_node.inputs():
+                        if i.debugName() in self._nid_to_node_name.keys():
+                            listconstr.append( \
+                                self._relay_map[self._nid_to_node_name[i.debugName()]])
+                        elif i.node().kind() == 'prim::Constant':
+                            listconstr.append(int(self._consts[i.debugName()]))
+                        elif i.debugName() in self._inputs_r.keys():
+                            listconstr.append(int(self._inputs_r[i.debugName()]))
+
+                    # Unwrap for tensors
+                    if len(listconstr) == 1:
+                        listconstr = listconstr[0]
+
+                    self._relay_map[nid] = listconstr
+                    self._nid_to_node_name[op_name] = nid
+                    nid = nid+1
+            else:
+                for i in op_node.inputs():
+                    if i.debugName() in self._nid_to_node_name.keys():
+                        for cnt in range(0, len(self._op_inputs_r[(op_name, operator)])):
+                            if isinstance(self._op_inputs_r[(op_name, operator)][cnt], str):
+                                if "call/var" in self._op_inputs_r[(op_name, operator)][cnt]:
+                                    self._op_inputs_r[(op_name, operator)][cnt] = \
+                                        self._relay_map[self._nid_to_node_name[i.debugName()]]
+                                    break
+
+                call = _convert_map[operator](self._op_inputs_r[(op_name, operator)],
+                                              self._op_inputs_types[(op_name, operator)])
+
+                self._relay_map[nid] = call
+                self._nid_to_node_name[op_name] = nid
+                nid = nid+1
+
+        outputs = []
+
+        for i in range(nid):
+            output = self._relay_map[i]
```

Comment:
I think self._relay_map can be a list, not a map.  You can literally replace self._relay_map with outputs.